### ISSUE TITLE
I bless you!

### ISSUE THREAD

**Issue by frederikhors** [bjorn3/rustc_codegen_cranelift#1372]
I open this issue that I will close as soon as you read it just to tell you that your project is AMAZING and please don't abandon it!

The Rust team should SERIOUSLY consider merging it IMMEDIATELY! Before anything else.

In even a relatively small project (about 20 domain entities, same crate) using normal cargo (rustc 1.69.0), the rebuild time is **20 seconds**.

Using cargo-clif (latest release, dfb1119) it drops to **11 seconds**!

Windows 10 x64.

All project build from scratch **2.20** minutes vs **1.05**!

**I bless you on each rebuild!** LOL! :smile:

**Comment by bjorn3** [comment]
Thanks for the praise!

A copy of cg_clif exists in the compiler/rustc_codegen_cranelift directory of the main rust repo which I occasionally sync. rust-lang/rust -> this repo every couple of days or so and this repo -> rust-lang/rust every 1-2 months or so. This copy needs to compile for rust's CI to pass, such that every time a compiler api changes, cg_clif needs to be fixed in the same PR. I'm working on additionally requiring the test suite of cg_clif to pass on rust's CI: https://github.com/bjorn3/rustc_codegen_cranelift/pull/1357 And after that there are a couple of other things I need to do before cg_clif can be shipped as rustup component on nightly: https://github.com/bjorn3/rustc_codegen_cranelift/milestone/2

tl;dr: I'm working on getting it shipped with rustc, but it might take a while.
